include/libopencm3/stm32/g0/adc.h

```diff
@@ -0,0 +1,327 @@
+#define ADC_CALFACT(adc)	MMIO32((adc) + 0xB4)
+
+/** ADC_OR Option register */
+#define ADC_OR(adc)			MMIO32((adc) + 0xD0)
+
+/* --- Register values -------------------------------------------------------*/
+
+/** @addtogroup adc_isr
+@{*/
+
+/** CCRDY: Channel Configuration Ready flag */
+#define ADC_ISR_CCRDY				(1 << 13)
+
+/**@}*/
+
+/** @addtogroup adc_ier
+@{*/
+
+/** CCRDYIE: Channel Configuration Ready Interrupt enable bit */
+#define ADC_IER_CCRDYIE				(1 << 13)
+
+/**@}*/
+
+/** @addtogroup adc_ccr
+@{*/
+
+#define ADC_CCR_PRESC_MASK		(0xf)
+#define ADC_CCR_PRESC_SHIFT		(18)
+/** @defgroup adc_ccr_presc ADC clock prescaler
+ *@{*/
+#define ADC_CCR_PRESC_NODIV		(0x0)
+#define ADC_CCR_PRESC_DIV1		(0x1)
+#define ADC_CCR_PRESC_DIV2		(0x2)
+#define ADC_CCR_PRESC_DIV6		(0x3)
+#define ADC_CCR_PRESC_DIV8		(0x4)
+#define ADC_CCR_PRESC_DIV10		(0x5)
+#define ADC_CCR_PRESC_DIV12		(0x6)
+#define ADC_CCR_PRESC_DIV16		(0x7)
+#define ADC_CCR_PRESC_DIV32		(0x8)
+#define ADC_CCR_PRESC_DIV64		(0x9)
+#define ADC_CCR_PRESC_DIV128	(0x10)
+#define ADC_CCR_PRESC_DIV256	(0x11)
+/**@}*/
+
+/**@}*/
+
+/** @addtogroup adc_cr
+@{*/
+
+/** ADVREGEN: Voltage regulator enable bit */
+#define ADC_CR_ADVREGEN				(1 << 28)
+
+/**@}*/
+
+/** @addtogroup adc_cfgr1
+@{*/
+
+/** CHSELRMOD: Mode Selection of the ADC_CHSELR register */
+#define ADC_CFGR1_CHSELRMOD			(1 << 21)
+
+/**@}*/
+
+/** @addtogroup adc_cfgr2
+@{*/
+
+#define ADC_CFGR2_CKMODE_SHIFT		(30)
+#define ADC_CFGR2_CKMODE_MASK		(0x3)
+/** @defgroup adc_cfgr2_ckmode ADC Clock mode
+ *@{*/
+#define ADC_CFGR2_CKMODE_ADCCLK		(0x0)
+#define ADC_CFGR2_CKMODE_PCLK_DIV2	(0x1)
+#define ADC_CFGR2_CKMODE_PCLK_DIV4	(0x2)
+#define ADC_CFGR2_CKMODE_PCLK		(0x3)
+/**@}*/
+
+/** LFTRIG: Low Frequency Trigger Mode enable bit */
+#define ADC_CFGR2_LFTRIG			(1 << 29)
+
+/** TOVS: Triggered Oversampling */
+#define ADC_CFGR2_TOVS				(1 << 9)
+
+#define ADC_CFGR2_OVSS_SHIFT		(5)
+#define ADC_CFGR2_OVSS_MASK			(0xf)
+/** @defgroup adc_cfgr2_ovss ADC Oversampling shift
+ *@{*/
+#define ADC_CFGR2_OVSS_BITS(bits)	(bits)
+/**@}*/
+
+#define ADC_CFGR2_OVSR_SHIFT		(2)
+#define ADC_CFGR2_OVSR_MASK			(0x7)
+/** @defgroup adc_cfgr2_ovsr ADC Oversampling ratio
+ *@{*/
+#define ADC_CFGR2_OVSR_2x (0x0)
+#define ADC_CFGR2_OVSR_4x (0x1)
+#define ADC_CFGR2_OVSR_8x (0x2)
+#define ADC_CFGR2_OVSR_16x (0x3)
+#define ADC_CFGR2_OVSR_32x (0x4)
+#define ADC_CFGR2_OVSR_64x (0x5)
+#define ADC_CFGR2_OVSR_128x (0x6)
+#define ADC_CFGR2_OVSR_256x (0x7)
+/**@}*/
+
+/** OVSE: Oversampler mode enable bit */
+#define ADC_CFGR2_OVSE				(1 << 0)
+
+/**@}*/
+
+/** @addtogroup adc_smpr
+@{*/
+
+/* SMP1 ADC Channel Sample Time selection */
+#define ADC_SMPR_SMPSEL_SHIFT			0x8
+#define ADC_SMPR_SMPSEL_MASK			0x7ffff
+#define ADC_SMPR_SMPSEL_CHANNEL_SHIFT(channel)	((channel) + ADC_SMPR_SMPSEL_SHIFT)
+#define ADC_SMPR_SMPSEL_CHANNEL_MASK			(1)
+/** @defgroup adc_smpr_smpsel ADC Sample Time selection
+@{*/
+#define ADC_SMPR_SMPSEL_SMP1	0x0
+#define ADC_SMPR_SMPSEL_SMP2	0x1
```

Comment:
oh boi, just read the refman on this stuff.  quite different from the other families :)